RecoHI/HiTracking/python/hiMixedTripletStep_cff.py

```diff
@@ -1,206 +1,159 @@
-    useRingSlector = cms.bool(True),
-    TTRHBuilder = cms.string('WithTrackAngle'),
-    clusterChargeCut = cms.PSet(refToPSet_ = cms.string('SiStripClusterChargeCutNone')),
-    minRing = cms.int32(1),
-    maxRing = cms.int32(1),
-    skipClusters = cms.InputTag('hiMixedTripletClusters')
-    )
-                                   )
-
-# SEEDS
-from RecoPixelVertexing.PixelTriplets.PixelTripletLargeTipGenerator_cfi import *
-PixelTripletLargeTipGenerator.extraHitRZtolerance = 0.0
-PixelTripletLargeTipGenerator.extraHitRPhitolerance = 0.0
-import RecoTracker.TkSeedGenerator.GlobalSeedsFromTriplets_cff
-hiMixedTripletSeedsA = RecoTracker.TkSeedGenerator.GlobalSeedsFromTriplets_cff.globalSeedsFromTriplets.clone()
-hiMixedTripletSeedsA.OrderedHitsFactoryPSet.SeedingLayers = 'hiMixedTripletSeedLayersA'
-hiMixedTripletSeedsA.OrderedHitsFactoryPSet.GeneratorPSet = cms.PSet(PixelTripletLargeTipGenerator)
-hiMixedTripletSeedsA.SeedCreatorPSet.ComponentName = 'SeedFromConsecutiveHitsTripletOnlyCreator'
-hiMixedTripletSeedsA.RegionFactoryPSet.RegionPSet.ptMin = 4.0
-hiMixedTripletSeedsA.RegionFactoryPSet.RegionPSet.originRadius = 0.005
-#hiMixedTripletSeedsA.RegionFactoryPSet.RegionPSet.nSigmaZ = 4.0
-hiMixedTripletSeedsA.RegionFactoryPSet.RegionPSet.originHalfLength = 10.0
-
-hiMixedTripletSeedsA.OrderedHitsFactoryPSet.GeneratorPSet.maxElement = 5000000
-hiMixedTripletSeedsA.ClusterCheckPSet.MaxNumberOfPixelClusters = 5000000
-hiMixedTripletSeedsA.ClusterCheckPSet.MaxNumberOfCosmicClusters = 50000000
+mixedTripletStepSeedLayersA.layerList = cms.vstring('BPix1+BPix2+BPix3', 'BPix1+BPix2+FPix1_pos', 'BPix1+BPix2+FPix1_neg', 
+                                        'BPix1+FPix1_pos+FPix2_pos', 'BPix1+FPix1_neg+FPix2_neg', 
+                                        'BPix2+FPix1_pos+FPix2_pos', 'BPix2+FPix1_neg+FPix2_neg')
+mixedTripletStepSeedLayersA.BPix.skipClusters = cms.InputTag('hiMixedTripletStepClusters')
+mixedTripletStepSeedLayersA.FPix.skipClusters = cms.InputTag('hiMixedTripletStepClusters')
+mixedTripletStepSeedLayersA.TEC.skipClusters = cms.InputTag('hiMixedTripletStepClusters')
+
+# TrackingRegion
+from RecoTracker.TkTrackingRegions.globalTrackingRegionWithVertices_cfi import globalTrackingRegionWithVertices as _globalTrackingRegionWithVertices
+hiMixedTripletStepTrackingRegionsA = _globalTrackingRegionWithVertices.clone(RegionPSet=dict(
+     precise = True,
+     useMultipleScattering = False,
+     beamSpot = "offlineBeamSpot",
+     useFoundVertices = True,
+     useFakeVertices       = False,
+     VertexCollection = "hiSelectedPixelVertex",
+     useFixedError = True,
+     fixedError = 3.75,#15.0
+     ptMin = 0.4,
+     originRadius = 1.5,
+     originRScaling4BigEvts = cms.bool(True),
+     halfLengthScaling4BigEvts = cms.bool(False),
+     ptMinScaling4BigEvts = cms.bool(True),
+     minOriginR = 0,
+     minHalfLength = 0,
+     maxPtMin = 0.7,
+     scalingStartNPix = 20000,
+     scalingEndNPix = 35000     
```

Comment:
more than a half of the above parameters are the same as in the original globalTrackingRegionWithVertices_cfi and should not really be repeated.
It may also be more practical to define also hiGlobalTrackingRegionWithVertices_cfi so that it can be reused directly instead of repeated copy-pastes of almost 20 lines

Reply:
It looks like a large fraction of the previous comment still applies
- halfLengthScaling4BigEvts has a common value with what's in the default globalTrackingRegionWithVertices_cfi. Was it really meant to be False?
- same for minOriginR
- same for minHalfLength

In case you find it complicated to define hiGlobalTrackingRegionWithVertices_cfi via PSet validation (fillDescriptions), please use the same code pattern as for hiMixedTripletStepClusters: set the "default" values here and then clone from it in other cases with modifications of only parameters that are changing

Reply:
The default parameters are now set in hiMixedTripletStep and then cloned into the other 2 iterations.